test/subscribers/publish_zync_event_subscriber_test.rb

```diff
@@ -1,14 +1,67 @@
+# frozen_string_literal: true
+
 require 'test_helper'
 
 class PublishZyncEventSubscriberTest < ActiveSupport::TestCase
-  def setup
-    @subscriber = PublishZyncEventSubscriber.new
+
+  class OIDCApplicationEventTest < ActiveSupport::TestCase
+    attr_reader :event, :subscriber, :publisher
+
+    setup do
+      service = FactoryBot.create(:simple_service, backend_version: 'oauth')
+      application = FactoryBot.create(:simple_cinstance, service: service)
+      @event = Applications::ApplicationCreatedEvent.create(application, nil)
+      @publisher = mock('publisher')
+      @subscriber = PublishZyncEventSubscriber.new(publisher)
+    end
+
+    test 'publish Zync Event for OIDC auth always' do
+      publisher.expects(:call).times(3).returns(:ok)
+
+      Rails.configuration.zync.stubs(skip_non_oidc_applications: false)
+      assert @subscriber.call(event)
+
+      Rails.configuration.zync.stubs(skip_non_oidc_applications: true)
+      assert @subscriber.call(event)
+
+      Rails.configuration.zync.stubs(skip_non_oidc_applications: nil)
+      assert @subscriber.call(event)
+    end
+
+    test 'do not publish Zync Event for non-OIDC auth only when not disabled' do
+      publisher.expects(:call).never
+      service = FactoryBot.create(:simple_service)
+      application = FactoryBot.create(:simple_cinstance, service: service)
+      event = Applications::ApplicationCreatedEvent.create(application, nil)
+      assert_nil @subscriber.call(event)
+    end
```

Comment:
The title is very confusing due to many negations, but I'd say this test belongs to the `NonOIDCApplicationEventTest` class.

Also, there's no test for "OIDC + setting not configured". Summarizing:

- OIDC
    - Skip Enabled -> :ok
    - Skip Disabled -> :ok
    - Not configured -> :ok <= **Doesn't exist**
- NoOIDC
    - Skip Enabled -> nil
    - Skip Disabled -> :ok
    - Not configured -> :ok <= **This test should be here**

Reply:
LOL, yeah :sweat_smile: I think it's a result of copy-paste and multiple refactors :grimacing:  I'll fix, thanks.

Reply:
Actually, this test is now redundant, because NoOIDC + "skip enabled" is covered by `'do not publish Zync Event if skipped'`